articles/best-practices/rules.md

```diff
@@ -1,247 +1,581 @@
 ---
 description: Best practices for writing and managing Auth0 Rules.
+toc: true
 topics:
   - best-practices
   - rules
-contentType:
-    - reference
+contentType: reference
 useCase:
   - best-practices
   - rules
 ---
 
 # Rules Best Practices
 
-This article covers best practices for using [rules](/rules). Before you start writing rules, we recommend that you:
+This article covers some best practices when using [rules](/rules). Rules can be used in a [variety of situations](/rules#what-can-i-use-rules-for-) as part of the pipeline where artifacts for authenticity are generated &mdash;i.e., an ID Token in [OpenID Connect (OIDC)](/protocols/oidc), an <dfn data-key="access-token">Access Token</dfn> in [OAuth 2.0](/protocols/oauth2), or an [assertion in SAML](/protocols/saml/saml-configuration/saml-assertions#use-rules). A new pipeline is created for each authentication request, in which rules execute.
```

Comment:
Last sentence. If the pipeline is what the rules execute in, then the comma is appropriate. If the authentication request is what the rules execute in, then it is not appropriate. Either way, I'd suggest simplifying this sentence a bit, as you're likely to have some user confusion either way.

Reply:
Updated content to remove ambiguity.